### PR TITLE
Correct Coil Sizing Summary report of Coil Total Capacity at Ideal Loads Peak

### LINKED ISSUE #9014: Coil Summary table values for capacity at ideal loads peak are incorrectly applying capacity as a function of temperature modification factor
Issue overview
--------------

@rraustad found that the capacity factors are being divided here when they should be multiplied.

https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/ReportCoilSelection.cc#L1233

https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/ReportCoilSelection.cc#L1499

Many coil sizing routines apply the capacity as a function of temperature to the peak load found during sizing so that the final autosize result has been cast back to the rating point.  The idea with the coil sizing summary table values is that the "At Ideal Loads Peak" values are as during sizing calculations and shouldn't be back cast to the rating point.  These lines of code are supposed to do that but had got it wrong.  The result is double counting of capacity factor.  

This issue makes the values for "Coil Total Capacity at Ideal Loads Peak [W]" , "Coil Sensible Capacity at Ideal Loads Peak [W]", and "Coil and Fan Net Total Capacity at Ideal Loads Peak [W]" incorrect. 

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes #9014
 -  Coil Sizing Summary report of Coil Total Capacity at Ideal Loads Peak [W] includes double counting of Capacity as a Function of Temperature Curve (DX Coils). This correction will also adjust calculated Coil Sensible Capacity at Ideal Loads Peak [W] and Coil and Fan Net Total Capacity at Ideal Loads Peak [W]. Changes will typically affect DX cooling coils.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] ~If changes fix a defect, the fix should be demonstrated in plots and descriptions~
 - [ ] ~If any defect files are updated to a more recent version, upload new versions here or on DevSupport~
 - [ ] ~If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label~
 - [ ] ~If structural output changes, add to output rules file and add OutputChange label~
 - [ ] ~If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies~

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/ReportCoilSelection.cc [hunk 1]
```diff
@@ -1230,7 +1230,7 @@ void ReportCoilSelection::setCoilCoolingCapacity(
     auto &c(coilSelectionDataObjs[index]);
     // no this is adjusted back to ratings    c->coilTotCapAtPeak = TotalCoolingCap;
     c->coilCapFTIdealPeak = coilCapFunTempFac;
-    c->coilTotCapAtPeak = TotalCoolingCap / c->coilCapFTIdealPeak;
+    c->coilTotCapAtPeak = TotalCoolingCap * c->coilCapFTIdealPeak;
```
Comment (rraustad) [inline src/EnergyPlus/ReportCoilSelection.cc:1233]: Coil sizing routines divide by CapFT at sizing conditions to provide the raw coil size required to meet the design load. The coil reports should multiply by CapFT to get back to the design load. Defect found while reviewing hand calculations of capacity in the Coil Sizing Summary.

**Comment by rraustad** [comment]
Expecting diffs only in html coil sizing reports.

File: tst/EnergyPlus/unit/ReportCoilSelection.unit.cc [hunk 2]
```diff
@@ -498,6 +499,11 @@ TEST_F(EnergyPlusFixture, ReportCoilSelection_ZoneEqCoil)
                                                                                 DXFlowPerCapMaxRatio);
     EXPECT_EQ(entAirDryBulbTemp, c1->coilDesEntTemp);
 
+    state->dataRptCoilSelection->coilSelectionReportObj->setZoneLatentLoadHeatingIdealPeak(zoneNum, zoneHeatingLatentLoad);
+    // Expect non-zero because this is a heating coil
+    EXPECT_LT(0.0, zoneHeatingLatentLoad);
+    EXPECT_NEAR(zoneHeatingLatentLoad, c1->rmLatentAtPeak, 0.000001);
+
```
Comment (rraustad) [inline tst/EnergyPlus/unit/ReportCoilSelection.unit.cc:506]: This unit test evaluated c1->rmLatentAtPeak prior to calling setCoilHeatingCapacity where zoneNum is allocated and initialized. The existing evaluation returned 0 as the rmLatentAtPeak as denoted in the comment. This new evaluation tests this call when it is able to execute properly and expects a non-zero result.

File: tst/EnergyPlus/unit/ReportCoilSelection.unit.cc [hunk 3]
```diff
@@ -528,4 +534,274 @@ TEST_F(EnergyPlusFixture, ReportCoilSelection_ZoneEqCoil)
...
+                                                                                curSysNum,
+                                                                                curZoneEqNum,
+                                                                                curOASysNum,
+                                                                                fanHeatGain,
+                                                                                coilCapFunTempFac,
+                                                                                DXFlowPerCapMinRatio,
+                                                                                DXFlowPerCapMaxRatio);
+    EXPECT_NEAR(RatedCoilTotCap * coilCapFunTempFac, c1->coilTotCapAtPeak, 0.000001);
+    EXPECT_LT(RatedCoilTotCap, c1->coilTotCapAtPeak, 0.000001);
+}
```
Comment (rraustad) [inline tst/EnergyPlus/unit/ReportCoilSelection.unit.cc:558]: New addition to existing unit test to test proper use of CapFT.

File: tst/EnergyPlus/unit/ReportCoilSelection.unit.cc [hunk 4]
```diff
@@ -528,4 +534,274 @@ TEST_F(EnergyPlusFixture, ReportCoilSelection_ZoneEqCoil)
...
+
+TEST_F(EnergyPlusFixture, ReportCoilSelection_ZoneEqCoolingCoil)
```
Comment (rraustad) [inline tst/EnergyPlus/unit/ReportCoilSelection.unit.cc:560]: Duplicate of unit test that evaluates reporting of heating coil, where here cooling is tested with similar rigor.

**Comment by rraustad** [comment]
Example correction using 5ZoneAutoDXVAV.

Develop shows 127224.61 Coil Total Capacity at Ideal Loads Peak [Btu/h]. This value was calculated by dividing the sized coil capacity by CapFT. To get back to the coil sizing result we multiply by CapFT, and to get back to the design load used to size this coil we multiply by CapFT again.

127224.61 * 1.0095 * 1.0095 = 129653.36. This branch shows 129658.82 with a difference of 5.46 due to rounding of reported CapFT. Debugger shows actual CapFT = 1.0095212 where this equation yields 129658.81, a match. The absolute diff shows 2434.21, 129658.82 - 127224.61 = 2434.21.

image

**Comment by rraustad** [comment]
@Myoldmopar @mjwitte @mitchute this should be good to go.

**Comment by rraustad** [comment]
The 1 small diff file is SolarShadingTestGPU which shows up now and then. No worries.

    -HeatTransfer Surface,ZN008:WALL002,Wall,,CTF - ConductionTransferFunction,EXTERIOR,0.644,0.588,,18.58,18.58,18.58,0.00,90.00,6.10,3.05,0.00,ExternalEnvironment,DOE-2,ASHRAETARP,SunExposed,WindExposed,0.50,0.50,0.65,0.35,4
    +HeatTransfer Surface,ZN008:WALL002,Wall,,CTF - ConductionTransferFunction,EXTERIOR,0.644,0.588,,18.58,18.58,18.58,0.00,90.00,6.10,3.05,0.00,ExternalEnvironment,DOE-2,ASHRAETARP,SunExposed,WindExposed,0.50,0.50,0.64,0.36,4

File: src/EnergyPlus/ReportCoilSelection.cc (hunk 1 above)
Comment (Myoldmopar) [reply src/EnergyPlus/ReportCoilSelection.cc:1233]: Great fix!

File: src/EnergyPlus/ReportCoilSelection.cc [hunk 5]
```diff
@@ -1496,7 +1496,7 @@ void ReportCoilSelection::setCoilHeatingCapacity(
     auto &c(coilSelectionDataObjs[index]);
     c->capIsAutosized = isAutoSize;
     c->coilCapFTIdealPeak = coilCapFunTempFac;
-    c->coilTotCapAtPeak = totalHeatingCap / c->coilCapFTIdealPeak;
+    c->coilTotCapAtPeak = totalHeatingCap * c->coilCapFTIdealPeak;
```
Comment (Myoldmopar) [inline src/EnergyPlus/ReportCoilSelection.cc:1499]: :+1:

File: tst/EnergyPlus/unit/ReportCoilSelection.unit.cc [hunk 6]
```diff
@@ -528,4 +534,274 @@ TEST_F(EnergyPlusFixture, ReportCoilSelection_ZoneEqCoil)
...
+                                                                                false,
+                                                                                curSysNum,
+                                                                                curZoneEqNum,
+                                                                                curOASysNum,
+                                                                                fanHeatGain,
+                                                                                coilCapFunTempFac,
+                                                                                DXFlowPerCapMinRatio,
+                                                                                DXFlowPerCapMaxRatio);
+    EXPECT_NEAR(RatedCoilTotCap * coilCapFunTempFac, c1->coilTotCapAtPeak, 0.000001);
+    EXPECT_LT(RatedCoilTotCap, c1->coilTotCapAtPeak);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ReportCoilSelection.unit.cc:806]: Nice.

**Review by Myoldmopar — APPROVED** [review]
This looks all good from code change standpoint.  Crazy how these things still hang out in the code.  :smile: I'll pull in latest develop and do some testing, but anticipate this merging very soon.

**Comment by Myoldmopar** [comment]
Built and ran fine locally.  Regressions locally also look just like before.  This is good to go in.  Thanks @rraustad !

**Pull request merged**